BAM-File-Transfer/BAM
# Uploading and Seeding via WebTorrent

# Changes
- Got [WebTorrent](https://webtorrent.io/) to be included into the client code.
- Added a "Send" Button that will generate the magnet link when clicked.
  - This magnet link will appear in the `console.log()`
  - Example:
  ```
  magnet:?xt=urn:btih:10c313a0be62fb86a00dac43f120b93f4c237f4c&dn=Scrum_combined.pdf&tr=udp%3A%2F%2Fexplodie.org%3A6969&tr=udp%3A%2F%2Ftracker.coppersurfer.tk%3A6969&tr=udp%3A%2F%2Ftracker.empire-js.us%3A1337&tr=udp%3A%2F%2Ftracker.leechers-paradise.org%3A6969&tr=udp%3A%2F%2Ftracker.opentrackr.org%3A1337&tr=wss%3A%2F%2Ftracker.btorrent.xyz&tr=wss%3A%2F%2Ftracker.fastcast.nz&tr=wss%3A%2F%2Ftracker.openwebtorrent.com
  ```
# How to test
1. Run the client code on `localhost:3000` using
```shell
npm run start
```
2. Click *Choose Files*.
3. Select the files you want to upload.
4. Click *Send*.
5. Open up the Console (press f12)
6. Copy (*Ctrl+C*) the magnet link.
7. Go to [Instant.io](https://instant.io/) and paste the magnet link there.
8. Press *Download*.
9. Verify that the downloaded file is correct.
